### PR TITLE
Add a step to regenerate custom certificates

### PR DESCRIPTION
**Pull request by adamlazik1**
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.4/Katello 4.6
* [x] Foreman 3.3/Katello 4.5
* [x] Foreman 3.2/Katello 4.4
* [x] Foreman 3.1/Katello 4.3

### CODE REVIEW

File: guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc [hunk 1]
```diff
@@ -42,16 +42,26 @@ For information on backups, see {AdministeringDocURL}Backing_Up_Server_and_Proxy
 
 ifdef::katello,satellite[]
 +
-. Regenerate certificates.
-On the main {Project} server:
+. Regenerate certificates on the main {ProjectServer}:
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc:45]:
```suggestion
. Regenerate certificates on your {ProjectServer}:
```

File: guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc [hunk 2]
```diff
@@ -42,16 +42,26 @@ For information on backups, see {AdministeringDocURL}Backing_Up_Server_and_Proxy
...
+.. Regenerate certificates for {SmartProxies} that use default certificates:
 +
 [options="nowrap" subs="attributes"]
 ----
 # {certs-generate} --foreman-proxy-fqdn "myproxy.example.com" \
-                       --certs-update-all \
-                       --certs-tar "~/myproxy.example.com-certs.tar"
+    --certs-update-all \
+    --certs-tar "~/myproxy.example.com-certs.tar"
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc:52]: Suggestions: use an attribute: https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-base.adoc#L97

**Review by maximiliankolb — COMMENTED** [review]
two minor suggestions, rest LGTM!

File: guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc (hunk 1 above)
Comment (adamlazik1) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc]: Thanks, aplied!

File: guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc (hunk 2 above)
Comment (adamlazik1) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc:52]: Applied.

**Comment by adamlazik1** [comment]
@maximiliankolb Thank you for your feedback. Incorporated the suggested changes.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc [hunk 3]
```diff
@@ -42,21 +42,31 @@ For information on backups, see {AdministeringDocURL}Backing_Up_Server_and_Proxy
...
+    --certs-update-all \
+    --certs-tar "~/{smartproxy-example-com}-certs.tar"
 ----
+.. Regenerate certificates for {SmartProxies} that use custom certificates:
 +
+[options="nowrap" subs="attributes"]
+----
+# {certs-generate} --foreman-proxy-fqdn "{smartproxy-example-com}" \
+    --certs-tar "~/{smartproxy-example-com}-certs.tar" \
+    --server-cert "/root/capsule_cert/capsule_cert.pem" \
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc:60]: Would you mind renaming "capsule_cert" to something suitable for all build targets? I'm thinking "proxy" in contrast to "server". Or maybe just reuse the attribute?

Comment (adamlazik1) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc]: @maximiliankolb I replaced `capsule_cert` with `{certs-proxy-context}_cert`. While doing this, I noticed two issues:
* orcharhino does not have `certs-proxy-context` attribute. Should we add one?
* orcharhino has the value of `smart-proxy-context` attribute set to `orcharhino Proxy`. I assume this is wrong for two reasons:
** Attribute `SmartProxy` has the exact same value for orcharhino
** `smart-proxy-context` is being used in anchors throughout the entire documentation

I would like to open another PR to change the value of `smart-proxy-context` for orcharhino, please let me know what should I change it to.

Comment (maximiliankolb) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc]: Thanks for spotting this! I think both of your suggestions to add the attribute and reword the other are good. If you add this in a separate commit to your branch, I'd be very happy. Else, I can of course provide a PR doing this later myself. Does this currently block your work? If so, ping me & I'll create a PR right away.

When merging, we can select "rebase and merge" to not squash your commits into one.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc [hunk 4]
```diff
@@ -42,21 +42,31 @@ For information on backups, see {AdministeringDocURL}Backing_Up_Server_and_Proxy
...
+. Regenerate certificates on your {ProjectServer}:
+.. Regenerate certificates for {SmartProxies} that use default certificates:
 +
 [options="nowrap" subs="attributes"]
 ----
-# {certs-generate} --foreman-proxy-fqdn "myproxy.example.com" \
-                       --certs-update-all \
-                       --certs-tar "~/myproxy.example.com-certs.tar"
+# {certs-generate} --foreman-proxy-fqdn "{smartproxy-example-com}" \
+    --certs-update-all \
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc:51]:
```suggestion
--certs-update-all \
```

IMHO most of the time, we have no leading whitespace.

**Review by maximiliankolb — COMMENTED** [review]
one suggestions on leading whitespace for commands, rest looks good to me!

File: guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc (hunk 4 above)
Comment (adamlazik1) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc]: Applied :+1:

File: guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc (hunk 3 above)
Comment (adamlazik1) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc]: I added the commit with updating the attributes but I need your help with what to rename them to. I set both `smart-proxy-context` and `certs-proxy-context` to `orcharhino-proxy`. Should I leave them I like this or rename to something else?

**Review by maximiliankolb — APPROVED** [review]
Thanks, LGTM!

**Comment by adamlazik1** [comment]
Are we good to merge?

**Comment by maximiliankolb** [comment]
Merged to master and cherry-picked: 2/4
   048555230..71a63c3bb  3.4 -> 3.4
   f9dda0aae..6c5365292  3.3 -> 3.3

@adamlazik1 I could not cherry-pick to 3.2 and 3.1. Since 3.2 is unsupported and not used by downstream products, you can create a single PR for 3.1.

**Pull request merged**